### PR TITLE
Build Improvements

### PR DESCRIPTION
**Pull request by alexlovelltroy**
* Add .goreleaser.yml
* Update Dockerfile to use goreleaser
* Use the version information more consistent with smd
* Use coredhcp-generator without cloning it.

### CODE REVIEW

File: .github/workflows/Realease.yaml
Comment (davidallendj) [inline .github/workflows/Realease.yaml:1]: Not really a big deal, but typo in the file name if it's meant to be "release.yaml".

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    main: ./cmd/
+    goos:
+      - linux
+    goarch:
+      - amd64
+      - arm64
+    goamd64:
+      - v3
+
+    # export GIT_STATE=$(if git diff-index --quiet HEAD --; then echo 'clean'; else echo 'dirty'; fi)
```
Comment (davidallendj) [inline .goreleaser.yml:22]: Can this be removed since the vars are defined in `Realease.yaml`?

File: internal/version/version.go [hunk 2]
```diff
@@ -1,7 +1,58 @@
...
+	fmt.Printf("Build Time: %s\n", BuildTime)
+	fmt.Printf("Git Branch: %s\n", GitBranch)
+	fmt.Printf("Git Tag: %s\n", GitTag)
+	fmt.Printf("Git State: %s\n", GitState)
+	fmt.Printf("Build Host: %s\n", BuildHost)
+	fmt.Printf("Go Version: %s\n", GoVersion)
+	fmt.Printf("Build User: %s\n", BuildUser)
+}
+
+func VersionInfo() string {
```
Comment (davidallendj) [inline internal/version/version.go:55]: Similar to comment from the [BSS PR](https://github.com/OpenCHAMI/bss/pull/48#discussion_r1822980267).

**Review by davidallendj — COMMENTED** [review]
I left two comments about the changes.

File: internal/version/version.go (hunk 2 above)
Comment (davidallendj) [reply internal/version/version.go:55]: Resolving based on https://github.com/OpenCHAMI/magellan/pull/66#issuecomment-2447810343.

**Comment by davidallendj** [comment]
I'm going to go ahead a approve since the comments aren't blocking.

**Review by davidallendj — APPROVED** [review]
Comments aren't blocking so approving.

File: .goreleaser.yml (hunk 1 above)
Comment (alexlovelltroy) [reply .goreleaser.yml:22]: I left it in as a comment so that people who are using goreleaser locally have a guide to follow.

**Review by synackd — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -32,6 +32,42 @@ This is meant to be built statically into
...
+### Building Locally with GoReleaser
+
+Once the environment variables are set, you can build the project locally using GoReleaser in snapshot mode (to avoid publishing).
+
+
+Follow the installation instructions from [GoReleaser’s documentation](https://goreleaser.com/install/).
+
+1. Run GoReleaser in snapshot mode with the --snapshot and --skip-publish flags to create a local build without attempting to release it:
+  ```bash
+  goreleaser release --snapshot --skip-publish --clean
```
[the remark this answers is not in the document]
Comment (synackd) [reply README.md:66]: Tested on cluster and `--skip-publish` wasn't recognized as an option but `--skip publush` was.

**Comment by alexlovelltroy** [comment]
just merged.  I'll fix the two typos and re-submit

**Pull request merged**